lib/learnosity/sdk/request/init.rb

```diff
@@ -3,226 +3,271 @@
+            end
+          when 'events'
+            hashed_users = {}
+            @request_packet['users'].each do |k, v|
+              hashed_users[k] =  hash_value(k + @secret)
+            end
+            @request_packet['users'] = hashed_users
+          when 'author', 'data'
+            @sign_request_data = true
+          else
+            raise Exception, "set_service_options() for #{@service} not implemented"
+          end
+        end
 
+        def generate_request_string
+          JSON.generate @request_packet unless request_packet.nil?
         end
 
-	def set_service_options()
-	  case @service
-	  when 'questions'
-	    # nothing to do
-	  when 'assess'
-	    if @request_packet.key?('questionsApiActivity')
-	      questionsApiActivity = @request_packet['questionsApiActivity']
-
-	      signatureParts = {}
-	      signatureParts['consumer_key'] \
-		= questionsApiActivity['consumer_key'] \
-		= @security_packet['consumer_key']
-
-	      signatureParts['domain'] = @security_packet['domain'] \
-		|| questionsApiActivity['domain'] \
-		|| 'assess.learnosity.com'
-
-	      signatureParts['timestamp'] \
-		= questionsApiActivity['timestamp'] \
-		= @security_packet['timestamp']
-
-	      signatureParts['expires'] = \
-		questionsApiActivity['expires'] \
-		= @security_packet['expires'] if @security_packet.key?('expires')
-
-	      signatureParts['user_id'] = \
-		questionsApiActivity['user_id'] = \
-		@security_packet['user_id']
-
-	      signatureParts['secret'] = @security_packet['secret']
-
-	      # Remove expires attribute if present but nil
-	      questionsApiActivity = hash_except(questionsApiActivity, 'expires') if questionsApiActivity['expires'].nil?
-
-	      @security_packet = signatureParts
-	      questionsApiActivity['signature'] = self.generate_signature()
-	      @request_packet['questionsApiActivity'] = questionsApiActivity
-
-	    end
-	  when 'items', 'reports'
-	    @sign_request_data = true
-	    if ! @request_packet.nil? and @request_packet.include? 'user_id' and
-		! @security_packet.include? 'user_id'
-	      @security_packet['user_id'] = @request_packet['user_id']
-	    end
-	  when 'events'
-	    hashedUsers = {}
-	    @request_packet['users'].each do |k, v|
-	      hashedUsers[k] =  self.hash_value(k + @secret)
-	    end
-	    @request_packet['users'] = hashedUsers
-	  when 'author', 'data'
-	    @sign_request_data = true
-	  else
-	    raise Exception, "set_service_options() for #{@service} not implemented"
-	  end
-	end
-
-	def generate_request_string()
-	  request_string = nil
-	  if ! @request_packet.nil?
-	    request_string = JSON.generate(@request_packet)
-	  end
-
-	  return request_string
-	end
-
-	def hash_value(value)
-	  return Digest::SHA256.hexdigest value
-	end
-
-	def hash_signature(signature_array)
-	  return self.hash_value(signature_array.join('_'))
-	end
+        def hash_value(value)
+		      Digest::SHA256.hexdigest value
```

Comment:
There seems to be a tab character in this line still.

Reply:
This is fixed in latest push